### DISCUSSION TITLE
context length slider wrongly configured (too high max context)

### DISCUSSION THREAD

**Discussion by Eve-146T** [open-webui/open-webui#7744] [Ideas]
you are unable to use the slider in the context length slider to set any meaningful context length, you have to manually enter a number. 
![image](https://github.com/user-attachments/assets/d6eedc6f-bb2f-4086-8909-a4a652266699)
as the middle is already 5 million context. something sensible would be that the max is 128k. 
![image](https://github.com/user-attachments/assets/63bbd25d-cdcd-4769-aa71-8413c5587544)

another option: the max context length changes on the model you are using:
gemini: 1m 
chatgpt: 128k 
llama: 128k 
qwen: 128k 
etc. 
if you manually click into it you would be able to change it to higher than that.

**Comment by Mushy-Snugglebites-badonkadonk** [comment]
There’s some scuttlebutt somewhere around the discussions here on GitHub or on the Discord server about someone complaining about the context slider and then the dev creating the outrageous range as a F.U. 

The sider is unusable as you pointed out. It would be great if the slider started at 2048 and increased by powers of two up to the model’s limit.

Here is the code in question:

https://github.com/open-webui/open-webui/blob/29a271959556743e6deb4d55a5a982983335d7ab/src/lib/components/chat/Settings/Advanced/AdvancedParams.svelte#L764

The step count needs to be modified so that the slider goes -1, 2048, 4096, 8192...128k in powers of two. 

Do let the community know how you’re able to run Gemini locally ;P

**Comment by Mushy-Snugglebites-badonkadonk** [comment]
A discussion on the Discord server about the outrageous slider:

https://discord.com/channels/1170866489302188073/1170866490174619690/1278329796010709032

**Comment by and1mon** [comment]
Maybe we could have a setting to specify our own step and max context?
